test/intl402/6.2.3_b.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2011-2012 Norbert Lindenberg. All rights reserved.
+// Copyright 2012  Mozilla Corporation. All rights reserved.
+// Copyright 2017 Microsoft Corporation. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 6.2.3
+description: Tests that language tags are canonicalized in return values.
+author: Norbert Lindenberg
+includes: [testIntl.js]
+---*/
+
+var canonicalizedTags = {
+    "de": ["de"],
+    "de-DE": ["de-DE", "de"],
+    "DE-de": ["de-DE", "de"],
+    "cmn": ["cmn"],
+    "CMN-hANS": ["cmn-Hans", "cmn"],
+    "cmn-hans-cn": ["cmn-Hans-CN", "cmn-Hans", "cmn"],
+    "es-419": ["es-419", "es"],
+    "es-419-u-nu-latn": ["es-419-u-nu-latn", "es-419", "es", "es-u-nu-latn"],
+    // -u-ca is incomplete, so it will not show up in resolvedOptions().locale
+    "cmn-hans-cn-u-ca-t-ca-x-t-u": ["cmn-Hans-CN-t-ca-u-ca-x-t-u", "cmn-Hans-CN-t-ca-x-t-u", "cmn-Hans-CN-t-ca-x-t", "cmn-Hans-CN-t-ca", "cmn-Hans-CN", "cmn-Hans", "cmn"],
+    "de-gregory-u-ca-gregory": ["de-gregory-u-ca-gregory", "de-gregory", "de-u-ca-gregory", "de"],
+    "no-nyn": ["nn"],
+    "i-klingon": ["tlh"],
+    "sgn-GR": ["gss"],
+    "ji": ["yi"],
+    "de-DD": ["de-DE", "de"],
+    "zh-hak-CN": ["hak-CN", "hak"],
+    "sgn-ils": ["ils"],
+    "in": ["id"],
+    "x-foo": ["x-foo"]
+};
+
+Object.getOwnPropertyNames(canonicalizedTags).forEach(function (tag) {
+    let locale = Intl.getCanonicalLocales(tag);
```

Comment:
While this test case seems legit, the place where this file is named and placed needs improvement.

`test/intl402/6.2.3.js` is using a legacy format where all the tests files were named after specs chapter references. That proved hard to maintain and consume as well. Unfortunately, renaming all the legacy file names is at a hard cost I cannot afford for now with time, but we're slowly reaching progress.

this file can be placed in the `test/intl402/Intl/getCanonicalLocales/` folder, the name should reflect what is checked here. The other files in the same folder should give a hint and what is done.